### ISSUE TITLE
Behavior of --p-flexible-reps-source flag

### ISSUE THREAD

**Issue by eelko09** [LadnerLab/q2-autopepsirf#26]
The --p-flexible-reps-source for qiime autopepsirf diffEnrich-tsv is intended to allow for samples with 1, 2, or 3+ replicates. There appears to be two issues:
1) currently it does not create enriched peptide files for single replicate samples and they also do not show up in the Qiime2 zenrich visualizations. Single replicate samples should still generate enriched files like how the "--p-s-enrich-source" flag works. 
2) the way it is written now when samples have 3+ replicates an enriched file is created for all pairwise combinations of the replicates. What would be better would be if all replicates were considered all the replicates together when determining enriched peptides. So each sample would have one enriched peptide output no matter how many replicates were included. 
I have attached a raw counts file below with samples with 1, 2, 3 and 4 replicates as well as the bin, negative matrix and example command for testing.
[Flexible_reps_source_test_files.zip](https://github.com/LadnerLab/q2-autopepsirf/files/10280564/Flexible_reps_source_test_files.zip)

**Comment by Num0Programmer** [comment]
I traced the output of autopepsirf - which is what is currently undesired - back to the Metadata class that Qiime2 uses. What led me there was line 209 in the file located here: https://github.com/LadnerLab/q2-autopepsirf/blob/main/q2_autopepsirf/actions/diffEnrich.py. I have concluded that this problem has more to do with how Qiime2 handles the replicate names than how PepSIRF does. As proof, I am uploading a few screenshots of a directory I created which shows that PepSIRF is able to enrich single replicate samples and samples with more than three replicates; this also shows that PepSIRF does look at all the replicates together rather than in pairs.

I do not underestimate this capability of PepSIRF was already known, so I have reached out to one of the engineers working on Qiime2 to start a conversation to address this issue. Until I hear back from the engineer, perhaps we can talk about possible work-arounds; or discuss other actions in regard to the priority of this issue, etc.

<img width="1328" alt="comboWR_raw_2mm_i1mm" src="https://user-images.githubusercontent.com/56742774/209875964-008ad6a3-d215-4904-8062-67d17aa2
<img width="713" alt="com
<img width="849" alt="10Z-HDI75_100CS_15000raw_enrich_module_test_out_dir" src="https://user-images.githubusercontent.com/56742774/209875984-2be72efa-69d4-499f-b774-3f55fd69826b.png">
boWR_raw_2mm_i1mm_PN" src="https://user-images.githubusercontent.com/56742774/209875979-b2aad0ba-3eec-402e-948d-a23359b40cd2.png">
3fe6.png">
